### ISSUE TITLE
Throwing a single optional parameters that the user didn't pass is used

### ISSUE THREAD

**Issue by fabiospampinato** [tinyhttp/tinyhttp#286]
**Describe the bug**

I have a route that looks like this:

```ts
app.get ( '/version/:channel?', async ( req, res ) => {} );
```

And the library throws if the optional parameter is not provided.

Stack trace:

```
(node:46178) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'length' of undefined
    at /Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:345:115
    at loop (/Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:357:86)
    at next (/Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:358:73)
    at /Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:130:5
    at /Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:145:13
    at /Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:354:40
    at loop (/Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:357:86)
    at App.handler (/Users/fabio/Projects/notable-website/node_modules/@tinyhttp/app/dist/index.cjs:359:9)
    at processImmediate (internal/timers.js:441:21)
```

Problematic line in question:

```js
const url = req$1.url.slice(req$1.url.indexOf(Object.values(params)[0]) + Object.values(params)[0].length);
```

Basically it seems to me like the library assumes that the value of the parameter is always a string, while in fact it seems like `req.getURLParams` can initialize parameters to `undefined` too.

**Expected behavior**

It should just work.

**Versions**

- `node`: 12.12.0
- `@tinyhttp/app`: @latest

**Comment by fabiospampinato** [comment]
Using something like this seems to fix that:

```js
(Object.values(params)[0] || '' )
```

Although I don't know, it seems like the types of `getURLParams` are wrong then? Also that looks like an expensive way to get the first parameter, it should be written in a more performant way.

**Comment by talentlessguy** [comment]
confirmed, bug is here: https://github.com/tinyhttp/tinyhttp/blob/08cff9e507ad8f4496a1181ea5e2b9d4baf50993/packages/app/src/app.ts#L315

**Comment by talentlessguy** [comment]
should be fixed in `1.3.12`

please check and say if you have any issues

**Comment by fabiospampinato** [comment]
I can confirm it's fixed, thanks for fixing this so quickly!
